### PR TITLE
Rosenbrock 3rd order perform_step! refactor

### PR DESCRIPTION
**Pull request by ParamThakkar123**
## Checklist

- [X] Appropriate tests were added
- [X] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [X] Any new documentation only uses public API
  
## Additional context

solves a part of #233

### CODE REVIEW

**Comment by ParamThakkar123** [comment]
@oscardssmith can you please review my latest commit ?? It gives me an error saying b is not defined. But I have defined it. Where did I made a mistake here ??

File: lib/OrdinaryDiffEqRosenbrock/src/rosenbrock_perform_step.jl [hunk 1]
```diff
@@ -1362,11 +840,21 @@ end
         @.. $(_vec(ks[stage]))=-linres.u
         integrator.stats.nsolve += 1
     end
+    du .= ks[end]
     u .+= ks[end]
 
+    @.. u=uprev + b[1] * ks[1] + b[2] * ks[2] + b[3] * ks[3]
```
Comment (oscardssmith) [inline lib/OrdinaryDiffEqRosenbrock/src/rosenbrock_perform_step.jl:846]: this is running even if  `b` isn't defined in the tableau.

Comment (ParamThakkar123) [reply lib/OrdinaryDiffEqRosenbrock/src/rosenbrock_perform_step.jl:846]: so what changes should be made here ??

**Comment by ParamThakkar123** [comment]
@oscardssmith can you please review ? Your reviews can help me move forward with this PR. So I can fix the errors and we can get this merged.

**Comment by ParamThakkar123** [comment]
> @oscardssmith can you please review ? Your reviews can help me move forward with this PR. So I can fix the errors and we can get this merged.

@ChrisRackauckas Can you please review this PR. ??

**Comment by ParamThakkar123** [comment]
@oscardssmith how the Rosenbrock32 and Rosenbrock23 be combined ?? Most of the test here failed because they are defined but maybe we need to combine them with the RosenbrockCache

**Comment by ParamThakkar123** [comment]
Rosenbrock32 and Rosenbrock23 have some extra fields in their caches compared to the RosenbrockCache

**Comment by oscardssmith** [comment]
Rosenbrock23 and 32 will need some extra work. Also you will need to make sure you're on a version of master new enough to include https://github.com/SciML/OrdinaryDiffEq.jl/pull/2307 since otherwise it will be totally broken.

**Comment by ParamThakkar123** [comment]
> Rosenbrock23 and 32 will need some extra work. Also you will need to make sure you're on a version of master new enough to include #2307 since otherwise it will be totally broken.

Okay. I need to pull #2307 from the upstream master in my branch

**Comment by ParamThakkar123** [comment]
image


@oscardssmith how do I fix this failure ??

**Pull request closed without merging**